### PR TITLE
Build spec document for OData Vocabularies

### PR DESCRIPTION
**Pull request by HeikoTheissen**
Subfolder `odata-vocabularies` contains the Markdown necessary for generating a spec document OData Vocabularies. It consists mostly of cross-references to the oasis-tcs/odata-vocabularies repo. This allows building the documents locally, but not in a GitHub action.

### CODE REVIEW

**Comment by ralfhandl** [comment]
Why not place all specs - including the Vocab spec - in the `odata-specs` repo?

We do already pull vocabs, schemas, and ABNF from other repos into the `odata-specs` repo when building the ZIP files.

**Comment by HeikoTheissen** [comment]
> Why not place all specs - including the Vocab spec - in the `odata-specs` repo?

Because then the source for the build step would come from a different repo, which is complicated to achieve in a GitHub action.

**Comment by ralfhandl** [comment]
> which is complicated to achieve in a GitHub action

Why would we want to do this in a GitHub action? We only need to run a build & zip script once we want to publish the document via OASIS, same as the other specs.

**Comment by HeikoTheissen** [comment]
@ralfhandl, what must go into the front matter, the introduction and the appendix?

**Comment by ralfhandl** [comment]
Same structure as the other specs

**Comment by ralfhandl** [comment]
References see https://docs.oasis-open.org/odata/odata-vocabularies/v4.0/csprd01/odata-vocabularies-v4.0-csprd01.html#_Toc472083027

**Comment by HeikoTheissen** [comment]
> References see https://docs.oasis-open.org/odata/odata-vocabularies/v4.0/csprd01/odata-vocabularies-v4.0-csprd01.html#_Toc472083027

The "Normative References" [OData-XXXVoc] make no sense, because the generated document already includes them completely.

File: odata-vocabularies/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+pagetitle: OData Vocabularies Version 4.0
+subtitle: Committee Specification Draft 02 / Public Review Draft 02
```
Comment (ralfhandl) [inline odata-vocabularies/meta.yaml:2]:
```suggestion
subtitle: Committee Specification Draft 02
```
Simplified terminology, see title pages of our 4.02 CSD01 docs

File: odata-vocabularies/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+filename: odata-vocabularies-v4.0-csprd02
+stage: csprd02
```
Comment (ralfhandl) [inline odata-vocabularies/meta.yaml:4]:
```suggestion
filename: odata-vocabularies-v4.0-csd02
stage: csd02
```

File: odata-vocabularies/Appendix.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+do we have considerations specific to URLs, for example length, encoding, privacy (use $batch if in doubt), ...?
+-->
+
+-------
+
+# Appendix ##asec Acknowledgments
+
+The contributions of the OASIS OData Technical Committee members, enumerated in [OData-Protocol](#ODataProtocol), are gratefully acknowledged.
+
```
Comment (ralfhandl) [inline odata-vocabularies/Appendix.md:52]:
```suggestion


## ##subasec Participants

**OData TC Members:**

| First Name | Last Name | Company |
| :--- | :--- | :--- |
| George | Ericson | Dell |
| Hubert | Heijkers | IBM |
| Ling | Jin | IBM |
| Stefan | Hagen | Individual |
| Michael | Pizzo | Microsoft |
| Christof | Sprenger | Microsoft |
| Ralf | Handl | SAP SE |
| Gerald | Krause | SAP SE |
| Heiko | Theißen | SAP SE |
| Martin | Zurmuehl | SAP SE |
```

File: odata-vocabularies/Appendix.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+| Hubert | Heijkers | IBM |
+| Ling | Jin | IBM |
+| Stefan | Hagen | Individual |
+| Michael | Pizzo | Microsoft |
+| Christof | Sprenger | Microsoft |
+| Ralf | Handl | SAP SE |
+| Gerald | Krause | SAP SE |
+| Heiko | Theißen | SAP SE |
+| Martin | Zurmuehl | SAP SE |
+-------
```
Comment (ralfhandl) [inline odata-vocabularies/Appendix.md:71]:
```suggestion

-------
```

File: lib/vocab2html.js
Comment (ralfhandl) [inline lib/vocab2html.js:1]: Links to the XML files need to be absolute URLs

**Review by ralfhandl — CHANGES_REQUESTED** [review]

**Comment by ralfhandl** [comment]
> @ralfhandl, what must go into the front matter, the introduction and the appendix?

Adjusted

File: lib/vocab2html.js
Comment (ralfhandl) [reply lib/vocab2html.js:1]: Ah, they are in the output HTML, just not in the live server

File: lib/build.js [hunk 5]
```diff
@@ -20,11 +21,12 @@ iterator(function (srcname, name, variant, meta) {
   var md = fs.createWriteStream(`${__dirname}/../docs/${name}/${name}.md`);
   var html = pandoc({
     "--metadata-file": `${__dirname}/../${srcname}/${variant}.yaml`,
+    ...meta.pandocOptions,
```
Comment (ralfhandl) [inline lib/build.js:24]: Something like this is missing in `server.js`

File: lib/vocab2html.js
Comment (HeikoTheissen) [reply lib/vocab2html.js:1]: They are, because of the final else branch.

File: lib/build.js (hunk 5 above)
Comment (ralfhandl) [reply lib/build.js:27]: Clicking on an XML link still shows
~~~
Cannot GET /odata-vocabularies/Org.OData.Core.V1.xml
~~~

Comment (HeikoTheissen) [reply lib/build.js:27]: Cannot reproduce: which link did you click?

Comment (ralfhandl) [reply lib/build.js:27]: Now it works 🤷🏻‍♂️ 😄

**Review by ralfhandl — APPROVED** [review]

File: lib/build.js (hunk 5 above)
Comment (HeikoTheissen) [reply lib/build.js:27]: I cannot reproduce that, the selector
```
a[href]:not([href^=http]):not([href^=\#])
```
finds nothing on odata-vocabularies.html.

Which link did you click?

Comment (HeikoTheissen) [reply lib/build.js:27]: Sorry, I cannot reproduce that, the CSS selector
```
a[href]:not([href^=http]):not([href^=\#])
```
finds nothing in odata-vocabularies.html.

**Review by ralfhandl — DISMISSED** [review]

**Review by ralfhandl — APPROVED** [review]

**Pull request merged**